pkg/controller/signer/signer-controller.go

```diff
@@ -118,6 +118,17 @@ func (r *ReconcileCSR) Reconcile(ctx context.Context, request reconcile.Request)
 		return reconcile.Result{}, nil
 	}
 
+	isValid, err := r.isValidUserName(ctx, csr.Spec.Username)
+	if err != nil {
+		return reconcile.Result{}, err
```

Comment:
enrich the error because look at the errors from the func, you may not tell where the error occured. Can you also enrich the List call - like failed to list nodes: %v

Reply:
sure, done.